src/compiler/checker.ts

```diff
@@ -8306,6 +8306,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 (getNodeLinks(context.enclosingDeclaration).fakeScopeForSignatureDeclaration || !findAncestor(node, n => n === context.enclosingDeclaration)) &&
                 !(sym && sym.flags & SymbolFlags.TypeParameter)
             ) {
+                sym = sym?.exportSymbol ?? sym;

```

Comment:
Should this be `getExportSymbolOfValueSymbolIfExported`?

Reply:
I believe that only works for value symbols not for type symbols since it checks `(symbol.flags & SymbolFlags.ExportValue)` and in the binder only value symbols get this flag:

```ts

const exportKind = symbolFlags & SymbolFlags.Value ? SymbolFlags.ExportValue : 0;
                const local = declareSymbol(container.locals, /*parent*/ undefined, node, exportKind, symbolExcludes);
                local.exportSymbol = declareSymbol(container.symbol.exports!, container.symbol, node, symbolFlags, symbolExcludes);

```

Or am I misunderstanding this?

Reply:
I swapped  this line for `sym = getExportSymbolOfValueSymbolIfExported(sym);` and nothing happened.

Truthfully I don't really know what I'm doing here, other than remembering past bugs where we forgot to call this function 😄

Reply:
Ah, I did some more debugging and understood it. The local symbol for types actually don't have any flags, so when you look them up you always get the export symbol. So there is never any need to look for the export symbol of a type. 

Made the change.